Eitri, Tyr & docker tests: Launch even if docker can't reach server

pbougue opened the pull request:
Allows to launch eitri or docker_test when not connected to internet.

This was hand-tested: launched `eitri` and `make docker_test` while offline.

TODO:

- [x] Check if we can even do smarter checks

Commit:
Eitri, Tyr & docker tests: Do not block launch when docker can't reach server

pbougue commented:
GitHub hooks are still messed-up, actual build : https://ci.navitia.io/job/navitia_pull_request_build/5182/consoleFull

nonifier commented on source/eitri/docker_wrapper.py:
```diff
@@ -65,15 +65,25 @@ def old_school_cnx_string(self):
-            log.debug(build_output)
+        log.info("Trying to build/update the docker image")
+        try:
+            for build_output in self.docker.build(POSTGIS_IMAGE, tag=POSTGIS_CONTAINER_NAME, rm=True):
```
Do we have to build the image every time ? Would it not be better to pull the image instead ?

pbougue replied:
Then we have to have a job building automatically the image, pushing it to a docker repo, and pull it in tests, if I'm not missing something.
It seems too big of a change to me.

Commit:
upgrade from docker-py to docker

warning: will need uninstalling docker-py as they are not compatible

also little improvments on checking everything is fine (image built before, container deleted after)

Commit:
--clang warning

pbougue commented:
Please read by commit.
Changed docker lib from https://pypi.org/project/docker-py/ to https://pypi.org/project/docker/ (it's the evolution of the lib, only it's maintained).

Warning: those 2 libs are not compatible, so you will have to remove `docker-py` from your virtualenv (the simplest is just to dump the venv).
It should not affect anything/anybody else than developpers, as it is only used by eitri, docker_test and tyr tests (none are launched on other places than dev's laptops, so far)

pbougue commented:
Jenkins, test this please
